### PR TITLE
EXIFOrientation Handling improvements

### LINKED ISSUE #1020: Picasso Exif Handling is incomplete
Picasso only handles the rotation and normal EXIF orientation settings correctly i.e.

ORIENTATION_NORMAL=1
ORIENTATION_ROTATE_180=3
ORIENTATION_ROTATE_270=8
ORIENTATION_ROTATE_90=6

It fails to handle 

ORIENTATION_FLIP_HORIZONTAL=2
ORIENTATION_FLIP_VERTICAL=4
ORIENTATION_TRANSPOSE=5
ORIENTATION_TRANSVERSE=7

Examples of all of these can be found at 

https://github.com/recurser/exif-orientation-examples

The fix is fairly straightforward but I can send in a pull request from a forked repository if preferred.

### PR DESCRIPTION
**Pull request by JohnWowUs**
closes #1020

### CODE REVIEW

File: picasso/src/main/java/com/squareup/picasso/FileRequestHandler.java [hunk 1]
```diff
@@ -45,15 +45,6 @@
...
-      case ORIENTATION_ROTATE_90:
-        return 90;
-      case ORIENTATION_ROTATE_180:
-        return 180;
-      case ORIENTATION_ROTATE_270:
-        return 270;
-      default:
-        return 0;
-    }
+    return orientation;
```
Comment (dnkoutso) [inline picasso/src/main/java/com/squareup/picasso/FileRequestHandler.java]: nit: inline return with the line above.

File: picasso/src/main/java/com/squareup/picasso/BitmapHunter.java [hunk 2]
```diff
@@ -552,8 +560,12 @@ static Bitmap transformResult(Request data, Bitmap result, int exifRotation) {
       }
     }
 
-    if (exifRotation != 0) {
-      matrix.preRotate(exifRotation);
+    if (exifOrientation != 0) {
+      matrix.preRotate(getExifRotation(exifOrientation));
+      int exifTranslation = getExifTranslation(exifOrientation) ;
+      if (exifTranslation != 1) {
+        matrix.postScale(exifTranslation,1) ;
```
Comment (dnkoutso) [inline picasso/src/main/java/com/squareup/picasso/BitmapHunter.java]: nit formatting is off here.

File: picasso/src/main/java/com/squareup/picasso/BitmapHunter.java [hunk 3]
```diff
@@ -570,4 +582,33 @@ private static boolean shouldResize(boolean onlyScaleDown, int inWidth, int inHe
...
     return !onlyScaleDown || inWidth > targetWidth || inHeight > targetHeight;
   }
+
+ static int getExifRotation(int orientation) {
+    switch (orientation) {
+      case ORIENTATION_ROTATE_90:
+      case ORIENTATION_TRANSPOSE:
+        return 90;
+      case ORIENTATION_ROTATE_180:
+      case ORIENTATION_FLIP_VERTICAL:
```
Comment (JakeWharton) [inline picasso/src/main/java/com/squareup/picasso/BitmapHunter.java]: A flip != 180 rotation

File: picasso/src/main/java/com/squareup/picasso/BitmapHunter.java [hunk 4]
```diff
@@ -570,4 +582,33 @@ private static boolean shouldResize(boolean onlyScaleDown, int inWidth, int inHe
...
+        return 180;
+      case ORIENTATION_ROTATE_270:
+      case ORIENTATION_TRANSVERSE:
+        return 270;
+      default:
+        return 0;
+    }
+  }
+
+ static int getExifTranslation(int orientation)  {
```
Comment (dnkoutso) [inline picasso/src/main/java/com/squareup/picasso/BitmapHunter.java]: nit: 2 spaces needed here.

File: picasso/src/main/java/com/squareup/picasso/BitmapHunter.java (hunk 3 above)
Comment (JakeWharton) [inline picasso/src/main/java/com/squareup/picasso/BitmapHunter.java]: Nevermind. I see what you did.

**Comment by dnkoutso** [comment]
```
/home/travis/build/square/picasso/picasso/src/main/java/com/squareup/picasso/BitmapHunter.java:31:15: Unused import - android.media.ExifInterface.ORIENTATION_NORMAL.
/home/travis/build/square/picasso/picasso/src/main/java/com/squareup/picasso/BitmapHunter.java:35:70: ';' is preceded with whitespace.
/home/travis/build/square/picasso/picasso/src/main/java/com/squareup/picasso/BitmapHunter.java:565:64: ';' is preceded with whitespace.
/home/travis/build/square/picasso/picasso/src/main/java/com/squareup/picasso/BitmapHunter.java:567:42: ',' is not followed by whitespace.
/home/travis/build/square/picasso/picasso/src/main/java/com/squareup/picasso/BitmapHunter.java:567:44: ';' is preceded with whitespace.
/home/travis/build/square/picasso/picasso/src/main/java/com/squareup/picasso/FileRequestHandler.java:25:15: Unused import - android.media.ExifInterface.ORIENTATION_ROTATE_180.
/home/travis/build/square/picasso/picasso/src/main/java/com/squareup/picasso/FileRequestHandler.java:26:15: Unused import - android.media.ExifInterface.ORIENTATION_ROTATE_270.
/home/travis/build/square/picasso/picasso/src/main/java/com/squareup/picasso/FileRequestHandler.java:27:15: Unused import - android.media.ExifInterface.ORIENTATION_ROTATE_90.```

File: picasso/src/main/java/com/squareup/picasso/BitmapHunter.java (hunk 3 above)
Comment (JohnWowUs) [inline picasso/src/main/java/com/squareup/picasso/BitmapHunter.java]: Double checked it and it works. There's perhaps a single matrix operation that could achieve this. I'll look into it.

**Comment by dnkoutso** [comment]
Please also squash to one commit.

**Comment by dnkoutso** [comment]
@JohnWowUs have you signed CLA?

**Comment by JohnWowUs** [comment]
I'd forgotten to do that. It's done now.

**Comment by JohnWowUs** [comment]
Some additional minor work might be needed to make this compatible with the fixes in PR #699 (center crop and exif).

**Pull request merged**